plugins/CircuitViewer/io/CircuitLoader.cpp

```diff
@@ -182,13 +227,12 @@ struct CircuitProperties
     Boxd boundingBox;
 };
 
-brain::GIDSet _getFilteredGIDs(const brain::Circuit& circuit,
-                               const std::string& target,
-                               const CircuitProperties& properties)
+brain::GIDSet _getFilteredGIDs(
+    const std::function<brion::GIDSet(const std::string&)>& resolver,
+    const brain::Circuit& circuit, const std::string& target,
+    const CircuitProperties& properties)
 {
-    const auto allGIDs = circuit.getRandomGIDs(properties.density / 100.0,
-                                               target, properties.randomSeed);
-
+    const auto allGIDs = resolver(target);
```

Comment:
I find giving `resolver` and `target` as input a bit strange. What about giving a `brion::GidSet&& allGIDs` as input instead?

Reply:
Notice that the resolver may be a brain::Simulation or a brain::Circuit depending on which type of config file was used for loading. This is not the case with BlueConfigs, but it's for SONATA.
Anyway, I have removed this filtering in a follow up commit, so I wouldn't pay much attention to this code at this point.